app/settings/developer-apps/create/controller.ts

```diff
@@ -1,23 +1,21 @@
 import Controller from '@ember/controller';
-import { action } from '@ember/object';
 import RouterService from '@ember/routing/router-service';
 import { inject as service } from '@ember/service';
-import { ChangesetDef } from 'ember-changeset/types';
 import Intl from 'ember-intl/services/intl';
 import Toast from 'ember-toastr/services/toast';
 
-import DeveloperApp from 'ember-osf-web/models/developer-app';
+import DeveloperAppModel from 'ember-osf-web/models/developer-app';
 import Analytics from 'ember-osf-web/services/analytics';
 
 export default class SettingsApplicationCreateController extends Controller {
     @service analytics!: Analytics;
     @service intl!: Intl;
     @service router!: RouterService;
     @service toast!: Toast;
+    developerApp!: DeveloperAppModel;
 
-    @action
-    onSave(developerApp: DeveloperApp & ChangesetDef) {
-        this.toast.success(this.intl.t('settings.developer-apps.created'));
-        this.router.transitionTo('settings.developer-apps.edit', developerApp.get('id'));
+    init() {
+        super.init();
+        this.developerApp = this.store.createRecord('developer-app');
```

Comment:
I am thinking `setUpController` may be a better fit for this use case. My understanding is Controller `init()` gets called once (only one instance of the controller exists in the application) and so next time this route is visited the init() does not run i.e `this.developerApp` is still same instance as last (might lead to errors within the store/ ember-data). `setUpController` runs every time the route is entered (allowing us to create a new instance of the `developer-app` on every visit.